### PR TITLE
Search by catalog number in user's collection.

### PR DESCRIPTION
**Pull request by mukeshk**
Addressed to  #1098

### CODE REVIEW

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java [hunk 1]
```diff
@@ -504,6 +507,17 @@ public String searchSeriesByCatalog(
 				series = Collections.emptyList();
 				break;
 		}
+		
+		boolean searchInCollection =(inCollection != null)?Boolean.TRUE:Boolean.FALSE;
+		
+		if( Features.SEARCH_IN_COLLECTION.isActive() && searchInCollection && 
+			SecurityContextHolder.getContext().getAuthentication().isAuthenticated()){
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: Please, use `@CurrentUser` instead.

Example: https://github.com/php-coder/mystamps/blob/4e398bcf101141eb0e562c18a54f853bd233ed61/src/main/java/ru/mystamps/web/feature/series/SeriesController.java#L181

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java [hunk 2]
```diff
@@ -470,6 +472,7 @@ public String processAskForm(
 	public String searchSeriesByCatalog(
 		@RequestParam(name = "catalogNumber", defaultValue = "") String catalogNumber,
 		@RequestParam(name = "catalogName", defaultValue = "") String catalogName,
+		@RequestParam(name = "inCollection", defaultValue = "" ,required = false) Boolean inCollection,
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: Should we use `defaultValue = "false"`? Also, it seems that if we have default value, then we shouldn't specify `required = false` (please, check the javadoc to fully ensure in that).

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java [hunk 3]
```diff
@@ -504,6 +507,17 @@ public String searchSeriesByCatalog(
 				series = Collections.emptyList();
 				break;
 		}
+		
+		boolean searchInCollection =(inCollection != null)?Boolean.TRUE:Boolean.FALSE;
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: I think that we won't need this, if `defaultValue = "false"` will work.

**Comment by mukeshk** [comment]
There is no default value, I think I can remove the default value.
As the nature of checkbox, they are sent as request param only when checked.

**Comment by mukeshk** [comment]
https://github.com/php-coder/mystamps/wiki/checkstyle - I missed it.

**Comment by php-coder** [comment]
> There is no default value, I think I can remove the default value.
> As the nature of checkbox, they are sent as request param only when checked.

This is attribute of the `@RequestParam` annotation and it doesn't relate to checkbox at all. It's about parameter that is present in a request or it isn't. In case there is no parameter with such name, the default value is used. So, this is what we need in our case.

**Comment by php-coder** [comment]
> https://github.com/php-coder/mystamps/wiki/checkstyle - I missed it.

Better later than never. But anyway, our bot won't let this to pass forward :)

**Comment by mukeshk** [comment]
Aha ! ok - I did a null check. I think your approach is better. Let me try.

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java (hunk 2 above)
Comment (mukeshk) [reply src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: `@RequestParam(name = "inCollection", defaultValue = "false") Boolean inCollection,
with
<input id="in-collection" name="inCollection" type="checkbox" value="true"/>`

This worked

**Comment by mukeshk** [comment]
please review.

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java (hunk 2 above)
Comment (php-coder) [reply src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: Why we need `value="true"` for checkbox?

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java [hunk 4]
```diff
@@ -470,10 +472,11 @@ public String processAskForm(
 	public String searchSeriesByCatalog(
 		@RequestParam(name = "catalogNumber", defaultValue = "") String catalogNumber,
 		@RequestParam(name = "catalogName", defaultValue = "") String catalogName,
+		@RequestParam(name = "inCollection", defaultValue = "false") Boolean inCollection,
+		@CurrentUser Integer currentUserId,
 		Model model,
 		Locale userLocale,
 		RedirectAttributes redirectAttributes) {
-
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: Please, add this line back as it was before (to reduce diff).

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java [hunk 5]
```diff
@@ -504,6 +507,19 @@ public String searchSeriesByCatalog(
 				series = Collections.emptyList();
 				break;
 		}
+		
+		//boolean searchInCollection =(inCollection != null)?Boolean.TRUE:Boolean.FALSE;
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: remove this comment, please

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java [hunk 6]
```diff
@@ -504,6 +507,19 @@ public String searchSeriesByCatalog(
...
+		if (Features.SEARCH_IN_COLLECTION.isActive()
+			&& inCollection
+			&& SecurityContextHolder.getContext().getAuthentication().isAuthenticated()) {
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: We can just check whether `currentUserId` is not null.

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java [hunk 7]
```diff
@@ -504,6 +507,19 @@ public String searchSeriesByCatalog(
...
+			series = series
+					.stream()
+					.filter(
+						e->collectionService.isSeriesInCollection(currentUserId, e.getId())
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: please, add spaces around `->`

File: src/main/webapp/WEB-INF/views/site/index.html [hunk 8]
```diff
@@ -184,7 +184,7 @@ <h4 class="panel-title" th:text="#{t_search_by_catalog}">Search by catalog</h4>
 								</div>
 								<!-- @todo #673 Add integration tests for search by a catalog in user's collection -->
 								<div class="form-group" sec:authorize="isAuthenticated()" togglz:active="SEARCH_IN_COLLECTION">
-									<input id="in-collection" name="inCollection" type="checkbox" />
+									<input id="in-collection" name="inCollection" type="checkbox" value="true"/>
```
Comment (php-coder) [inline src/main/webapp/WEB-INF/views/site/index.html]: It should work without `value=true`

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java (hunk 4 above)
Comment (mukeshk) [reply src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: fixed.

Comment (php-coder) [reply src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: yeah, but I still see the diff :-( could you also add tab here?

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java (hunk 6 above)
Comment (mukeshk) [reply src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: ok

File: src/main/webapp/WEB-INF/views/site/index.html (hunk 8 above)
Comment (mukeshk) [reply src/main/webapp/WEB-INF/views/site/index.html]: Yes. verified. I thought with HTTP request param value should come as true, which spring would parse for the boolean.

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java (hunk 5 above)
Comment (mukeshk) [reply src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: removed.

**Comment by php-coder** [comment]
Could you, please, edit the first line of the commit message to be:
```
task(/series/search/by_catalog): implement the simplest search by catalog number in user's collection.
```

**Comment by mukeshk** [comment]
Some import series test cases are broken at the moment. 
Not sure if it's due to this pull.

**Comment by php-coder** [comment]
> Some import series test cases are broken at the moment.

No, it's the test cases. It's from PMD (see also https://github.com/php-coder/mystamps/wiki/pmd-cpd):

>[INFO] PMD Failure: ru.mystamps.web.feature.series.SeriesController:474 Rule:AvoidPrefixingMethodParameters Priority:4 Avoid prefixing parameters by in, out or inOut. Uses Javadoc to document this behavior..

But don't worry about it -- I'll disable this check from PMD today.

**Comment by mukeshk** [comment]
Added tab on the line.  479.

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java [hunk 9]
```diff
@@ -504,6 +507,18 @@ public String searchSeriesByCatalog(
 				series = Collections.emptyList();
 				break;
 		}
+		
+		if (Features.SEARCH_IN_COLLECTION.isActive()
```
Comment (php-coder) [inline src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: P.S. Could you add a comment before this `if``:
```
// @todo #1100 Optimize a search within user's collection
```

Comment (php-coder) [reply src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: Correction, there was a wrong number:

```
// @todo #1098 Optimize a search within user's collection
```

**Comment by php-coder** [comment]
Thank you @mukeshk !

File: src/main/java/ru/mystamps/web/feature/series/SeriesController.java (hunk 9 above)
Comment (php-coder) [reply src/main/java/ru/mystamps/web/feature/series/SeriesController.java]: I've added it in 742b7d8009cbbbec3f49b78207e186a3649ac517 commit.

**Pull request merged**